CONTRIBUTING.md

```diff
@@ -96,6 +96,64 @@ Closes #123
 
 In this case 123 is the corresponding issue number.
 
+### When End-To-End Tests Fail
+
+Every pull request triggers a full run of testing in the CI system.
+The failures reported by the code style checker (aka the "linter") and the unit tests are usually
+clear and easy to fix (and can be avoided by running `npm t` locally before creating a commit).
+But when an integration, or e2e test, fails, it's sometimes useful to consult the log files
+for the run.
+
+Here's how to see more details for a failed run:
+
+1. The test results can be found by first going to the GitHub page for the PR (for an example, find an open PR at https://github.com/rancher-sandbox/rancher-desktop/pulls?q=is%3Aopen+is%3Apr),
+then look for the list of checks, find the `E2E/Integration Tests` entry, and click on the `Details` link for the first view.
+
+![Failing E2E tests link](./docs/assets/images/cilogs/screenshot01.png)
+
+This takes you to a more detailed view of the test, and which tests failed, but the log files are a few more clicks away.
+
+2. Scroll to the bottom and click on the link that reads `View more details on Cirrus CI`.
+
+![Link to Cirrus CI](./docs/assets/images/cilogs/screenshot02.png)
+3. Once the full test run has completed, the `Artifacts` button acts like a file-system browser.
+   The key here is to *not* click on the download button, but to click anywhere else on the row containing the button.
+
+![artifacts expansion button](./docs/assets/images/cilogs/screenshot03.png)
+4. This opens up a file browser. Now click on the list-view icon (the left of the two icons at the right of the row).
+
+
+![list-view button](./docs/assets/images/cilogs/screenshot04.png)
+5. Find the `.tar` file for the failed test, and click on it to download it into your designated `downloads` directory.
+
+
+![tar-file link](./docs/assets/images/cilogs/screenshot05.png)
+
+6. Because each test's tar file contains the same log files, it's a good idea to extract them into a separate
+directory. For example, if you're working with PR 5188:
```

Comment:
Hmm, 5188 doesn't exist yet — hopefully we don't get a troll to make that PR with a giant shock image or something.  Perhaps change it to refer to PR 2672 (this one) instead?

Reply:
good choice